plugin/core/configurations.py

```diff
@@ -73,6 +80,26 @@ def disable_config(self, config_name: str, only_for_session: bool = False) -> No
             disable_in_project(self._window, config_name)
         self.update(config_name)
 
+    def record_crash(self, config_name: str) -> bool:
+        """
+        Signal that a session has crashed.
+
+        Returns True if the session should be restarted automatically.
+        """
+        now = datetime.now()
+        if config_name not in self._crashes:
+            self._crashes[config_name] = deque([now], maxlen=RETRY_MAX_COUNT)
+        else:
+            self._crashes[config_name].append(now)
+        timeout = now - RETRY_COUNT_TIMEDELTA
+        crash_count = len([crash for crash in self._crashes[config_name] if crash > timeout])
+        msg = "".join((
+            "session for config {} crashed. ",
+            "({} / {} times in the last {} seconds)"
+        )).format(config_name, crash_count, RETRY_MAX_COUNT, RETRY_COUNT_TIMEDELTA.total_seconds())
+        debug(msg)
```

Comment:
I would want exception included in the debug log.

And also a question of whether it should be `debug` or a normal log... Most people don't run with debug logging and having server crash without any feedback (even if only in the console) can result in people not understanding what happened and would likely also make support harder. We won't be able to just ask: "do you see any crash messages in the console" if someone asks why server resets sometimes?

Reply:
Agree! Pushed an update